packages/sdks-tests/src/e2e-tests/ab-test.spec.ts

```diff
@@ -45,7 +45,7 @@ const initializeAbTest = async (
   if (!baseURL) throw new Error('Missing baseURL');
 
   // SSR A/B tests do not seem to work on old NextJS. Likely a config issue.
-  test.skip(packageName === 'gen1-next');
+  test.skip(packageName === 'gen1-next-14' || packageName === 'gen1-next-15');
```

Comment:
from what i found, if we don't set `builder.canTrack` as false ssr a/b tests work

Reply:
Ahh, good find! so this:

https://github.com/BuilderIO/builder/blob/e77aae4d8ad29fb03c65c8cb51a0add7f3c17c0b/packages/react-tests/nextjs/pages/[[...index]].tsx#L36-L38

is breaking SSR tests. We can probably fix it then by including SSR tests in this clause: 

https://github.com/BuilderIO/builder/blob/e77aae4d8ad29fb03c65c8cb51a0add7f3c17c0b/packages/react-tests/nextjs/pages/[[...index]].tsx#L46-L51

Reply:
ahh makes sense now on why the call was still happening, as we use it inside useEffect, the first track fires off before reaching that check. Have updated the builder.canTrack calls to happen on global level before component initialization which fixes our issue